### PR TITLE
[ARM] Add missing earlyclobber to sqrshr and uqrshl instructions.

### PR DESCRIPTION
**Pull request by AlfieRichardsArm**
This avoids possible undefined behavior using the same register for Rm and Rda.

Additionally adds a check in MC to produce an error upon parsing this case.

### CODE REVIEW

**Review by jthackray — APPROVED** [review]
LGTM.

**Comment by AlfieRichardsArm** [comment]
@jthackray Updated this after I worked out how to run clang-format properly

**Review by jthackray — APPROVED** [review]
LGTM

File: llvm/test/CodeGen/AArch64/sqrshr-uqrshl-unpredictable.ll [hunk 1]
```diff
@@ -0,0 +1,21 @@
+; RUN: llc -mtriple armv8.1m.main -mattr=+mve %s -o - | FileCheck %s
```
Comment (davemgreen) [inline llvm/test/CodeGen/AArch64/sqrshr-uqrshl-unpredictable.ll:1]: This test should be under CodeGen/Arm or CodeGen/Thumb2, as it's not AArch64.

I would also personally use the update_llc_test_checks script to generate the check lines.

Comment (AlfieRichardsArm) [reply llvm/test/CodeGen/AArch64/sqrshr-uqrshl-unpredictable.ll]: Thank you, I have moved the test

Unfortunately I couldn't get the update_llc_test_checks took to work, I ran into the same issue as https://github.com/llvm/llvm-project/issues/66409

File: llvm/test/MC/AArch64/mve-sqrshr-uqrshl-earlyclobber.s [hunk 2]
```diff
@@ -0,0 +1,6 @@
+@ RUN: not llvm-mc -triple armv8.1m.main -mattr=+mve < %s 2>&1 | FileCheck %s
```
Comment (davemgreen) [inline llvm/test/MC/AArch64/mve-sqrshr-uqrshl-earlyclobber.s:1]: This test can be moved too :)

**Review by davemgreen — COMMENTED** [review]
That sounds OK. The test is probably good as-is. If you can move the other test then this LGTM too. Thanks

File: llvm/test/MC/AArch64/mve-sqrshr-uqrshl-earlyclobber.s (hunk 2 above)
Comment (AlfieRichardsArm) [reply llvm/test/MC/AArch64/mve-sqrshr-uqrshl-earlyclobber.s]: Oh I just checked this and seemingly my eyes skipped right over it. Thank you!

**Review by davemgreen — APPROVED** [review]
Thanks. LGTM

**Pull request merged**